### ISSUE TITLE
Check if Terminal.app has FDA

### ISSUE THREAD

**Issue by SynAckJack** [SynAckJack/macOS-ir#3]
Due to the data that is collected, `collect.sh` needs Full Disk Access to function correctly.

Is there a standard way to check if a specific app has FDA? Possibly try access an area that requires FDA and check.

**Comment by 0xmachos** [comment]
I don't believe there's a standard way to check Full Disk Access (FDA) via shell scripts. 

You could invoke `sqlite3`on `TCC.db` to check the FDA entry for `Terminal.app`. 

In mOSL I settled on checking if `$HOME/Library/Mail` is readable. 

```
function full_disk_access_check {

  if [ -r "$HOME/Library/Mail" ]; then
    return 0
  else
    return 1
  fi
}
```
- [Source](https://github.com/0xmachos/mOSL/blob/db527e68026f32e498a38bf47144c4a9e9b7209c/Lockdown#L339-L346)

Mail does't have a specific entry in Privacy Settings to allow access so you need FDA to read it. This guarantees no false positives, until Apple changes TCC again...

**Comment by SynAckJack** [comment]
Implemented `check_fda` in c068c7be6b1663b9238a7740cd3d79438c7857da and merges in #6.

Closing issue.
